src/service/setupService.ts

```diff
@@ -159,40 +159,151 @@ export class SetupService {
         return pythonExecutablePath;
     };
 
-    public getCurrentPythonExecutablePath = async () => {
-        let originalpythonExecutablePath = "";
-
+    public getCurrentPythonExecutablePath = async (
+        isTryingPython3: boolean = false
+    ) => {
+        let originalPythonExecutablePath = "";
+        const systemPythonVar = isTryingPython3
+            ? GLOBAL_ENV_VARS.PYTHON3
+            : GLOBAL_ENV_VARS.PYTHON;
         // try to get name from interpreter
         try {
-            originalpythonExecutablePath = getConfig(CONFIG.PYTHON_PATH);
+            originalPythonExecutablePath = getConfig(CONFIG.PYTHON_PATH);
         } catch (err) {
-            originalpythonExecutablePath = GLOBAL_ENV_VARS.PYTHON;
+            originalPythonExecutablePath = systemPythonVar;
         }

```

Comment:
Honestly, I would just prompt the user to pick the interpreter in the python extension if nothing is selected and call it a day.
That way you have a single source of truth, we piggyback on any improvements that the python extension might introduce in the future and we keep this method much simpler.
What do you think?

Reply:
It's just that unix systems sometimes have an issue where the interpreter lists a different version in the bottom left corner than there is actually in the settings. For example, if you don't set a python path, the value in settings defaults to "python". In the extension, it will show your newest version of Python, but "python" will point to python2.7. To mitigate the confusion of people thinking that they're on the right version while they haven't even selected an interpreter, I put this in.

Reply:
https://github.com/microsoft/vscode-python-devicesimulator/issues/163#issuecomment-595390192

Reply:
This was requested by Stephane.

Reply:
Thanks for clarifying!